sorts/tree_sort_2.py

```diff
@@ -0,0 +1,138 @@
+"""
+
+tree_sort_2
+
+- makes use of the data structure Binary Tree to sort the list in O(nlogn) time.
+- Binary Search Tree [BST]
+  - For each node with value v
+    - All values in the left subtree are < v
+    - All values in the right subtree are > v
+  - No Duplicate values.
+- Logic:
+  - Build a Binary Search Tree
+  - Traverse the tree in increasing order.
+
+"""
+from typing import Any
+
+
+class TreeSort:
+    """
+    Binary Search Tree Data Structure
+        - An empty left and right branches will be created for every value inserted,
+        to perform better in recursive methods
+    """
+
+    # Constructor
+    def __init__(self, initval: Any = None) -> None:
+        self.value: Any
+        self.left: Any
+        self.right: Any
+
+        self.value = initval
+        if self.value:
+            self.left = TreeSort()
+            self.right = TreeSort()
+        else:
+            self.left = None
+            self.right = None
+        return
+
+    def isempty(self) -> bool:
+        """
+        Empty nodes are None valued.
+        Returns true if tree is empty else False
+        >>> tree = TreeSort()
+        >>> tree.isempty()
+        True
+        >>> tree.insert(1)
+        >>> tree.isempty()
+        False
+        """
+        if self.value is None:
+            return True
+        return False
```

Comment:
```suggestion
        return self.value is None
```

Reply:
No. This one, if we do like this. It gets rejected by mypy.